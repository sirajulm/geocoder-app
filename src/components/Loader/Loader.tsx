import React from 'react';
import styled from 'styled-components';
import spinnerImage from '../../images/spin.svg'

const StyledOverlay = styled.div`
    position: absolute;
    top: 0;
    bottom: 0;
    right: 0;
    left: 0;
    z-index: 999;
    background-color: rgba(0,0,0, 0.7);
`;

const StyledSpinnerWrapper = styled.div`
    width: 100px;
    margin: auto;
    position: relative;
    top: 50%;
    margin-top: -50px;
`;
const StyledSpinner = styled.img`
`;
const Loader = () => {
    return <StyledOverlay>
        <StyledSpinnerWrapper>
            <StyledSpinner src={spinnerImage} alt="spinner" />
        </StyledSpinnerWrapper>
    </StyledOverlay>
}

export default Loader;