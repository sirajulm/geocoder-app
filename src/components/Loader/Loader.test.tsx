import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer'
import Loader from './Loader';


describe('Loader', () => {
    it('should render', () => {
        const component = renderer.create(<Loader />).toJSON()
        expect(component).toMatchSnapshot()
    })
})