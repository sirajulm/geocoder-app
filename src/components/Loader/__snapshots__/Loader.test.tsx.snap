// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render 1`] = `
<div
  className="sc-bdVaJa kbGhZi"
>
  <div
    className="sc-bwzfXH iazbFD"
  >
    <img
      alt="spinner"
      className="sc-htpNat hHtpcc"
      src="spin.svg"
    />
  </div>
</div>
`;
